lib/zone.ts

```diff
@@ -1294,7 +1318,13 @@ const Zone: ZoneType = (function(global: any) {
     scheduleMicroTask: scheduleMicroTask,
     showUncaughtError: () => !(Zone as any)[__symbol__('ignoreConsoleErrorUncaughtError')],
     patchEventTargetMethods: () => false,
-    patchOnProperties: noop
+    patchOnProperties: noop,
+    beforeRunTask: (zone: Zone, task: Task) => {
```

Comment:
Do you mean we should move the `beforeRunTask` and `afterRunTask` from class Zone so they will not be accessible from outside such as `zone.beforeRunTask`?